### PR TITLE
Fk cloud slideseq

### PR DESCRIPTION
**Pull request by nikellepetrillo**
### Description

PR containing the first release of the SlideSeq pipeline. 

----

### Checklist 
If you can answer "yes" to the following items, please add a checkmark next to the appropriate checklist item(s) **and** notify our WARP documentation team by tagging either @ekiernan or @kayleemathews in a comment on this PR.

- [x] Did you add inputs, outputs, or tasks to a workflow?
- [x] Did you modify, delete or move: file paths, file names, input names, output names, or task names?
- [x] If you made a changelog update, did you update the pipeline version number?

### CODE REVIEW

File: dockers/skylab/pytools/tools/create_loom_slide_seq.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
+import argparse
```
Comment (khajoue2) [inline dockers/skylab/pytools/tools/create_loom_slide_seq.py:1]: This script is not used in this version of the pipeline. Not sure if we want to keep it or leave it on the branch.

File: pipelines/skylab/slideseq/test_inputs/Scientific/Puck_200501_23.mm10.fastqs.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (khajoue2) [inline pipelines/skylab/slideseq/test_inputs/Scientific/Puck_200501_23.mm10.fastqs.json:1]: We do not want to keep most of the scientific test jsons. Only one should be there. See ticket PD-2092 on jira.

**Review by khajoue2 — CHANGES_REQUESTED** [review]

File: pipelines/skylab/slideseq/test_inputs/Plumbing/slideseq_downsample.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
```
Comment (khajoue2) [inline pipelines/skylab/slideseq/test_inputs/Plumbing/slideseq_downsample.py:1]: Do we want to keep this script here or under pytools dockers? (ultimately as part of warp-tools repo)

**Review by khajoue2 — CHANGES_REQUESTED** [review]

File: pipelines/skylab/slideseq/test_inputs/Plumbing/slideseq_downsample.py (hunk 3 above)
Comment (ekiernan) [reply pipelines/skylab/slideseq/test_inputs/Plumbing/slideseq_downsample.py:1]: @khajoue2 I guess my thought is that this script is very specific to the plumbing test inputs, which isn't exactly necessary for the pipeline itself. I think it makes sense to have the script that was used to make the data sit next to the data itself. But if we decide to move it, we just need to be clear where to find it.

File: pipelines/skylab/slideseq/test_inputs/Scientific/Puck_210817_11.mm10.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "SlideSeq.r1_fastq": [
```
Comment (ekiernan) [inline pipelines/skylab/slideseq/test_inputs/Scientific/Puck_210817_11.mm10.json:2]: these are all in the hca-dcp-mint storage.

File: dockers/skylab/pytools/tools/create_loom_slide_seq.py (hunk 1 above)
Comment (nikellepetrillo) [reply dockers/skylab/pytools/tools/create_loom_slide_seq.py:1]: Hi @khajoue2, it looks like this script is in the pytools dir. I am good with keeping it here if you are!

File: pipelines/skylab/slideseq/test_inputs/Scientific/Puck_210817_11.mm10.json (hunk 4 above)
Comment (nikellepetrillo) [reply pipelines/skylab/slideseq/test_inputs/Scientific/Puck_210817_11.mm10.json:2]: ty @ekiernan! i believe @aawdeh will be replacing the hca-dcp-mint paths with broad-gtoc-test-storage paths in this ticket https://broadworkbench.atlassian.net/browse/PD-2092. I can update this branch once that gets merged into develop!

File: .dockstore.yml [hunk 5]
```diff
@@ -101,4 +101,8 @@ workflows:
 
   - name: BuildIndices
     subclass: WDL
-    primaryDescriptorPath: /pipelines/skylab/build_indices/BuildIndices.wdl
\ No newline at end of file
+    primaryDescriptorPath: /pipelines/skylab/build_indices/BuildIndices.wdl
+
+  - name: Slideseq
```
Comment (ekiernan) [inline .dockstore.yml:106]:
```suggestion
- name: SlideSeq
```

File: pipelines/skylab/slideseq/test_inputs/Scientific/Puck_210817_11.mm10.json (hunk 4 above)
Comment (aawdeh) [reply pipelines/skylab/slideseq/test_inputs/Scientific/Puck_210817_11.mm10.json:2]: The file paths are now updated.

Comment (nikellepetrillo) [reply pipelines/skylab/slideseq/test_inputs/Scientific/Puck_210817_11.mm10.json:2]: @aawdeh I am still seeing the paths point to hca-dcp-mint-test-data, do you need to push up another commit?

Comment (nikellepetrillo) [reply pipelines/skylab/slideseq/test_inputs/Scientific/Puck_210817_11.mm10.json:2]: @aawdeh looks good! 👍 i just needed to refresh

**Review by khajoue2 — APPROVED** [review]

File: pipelines/skylab/slideseq/SlideSeq.wdl [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    }
+
+    call StarAlign.MergeStarOutput as MergeStarOutputs {
+        input:
+            barcodes = STARsoloFastqSlideSeq.barcodes,
+            features = STARsoloFastqSlideSeq.features,
+            matrix = STARsoloFastqSlideSeq.matrix,
+            input_id = input_id
+    }
+    if ( !count_exons ) {
```
Comment (ekiernan) [inline pipelines/skylab/slideseq/SlideSeq.wdl:113]: @khajoue2 It looks like the bead barcodes are being added to the Loom if the count exons option is false? Do we want the pipeline to behave like this or do we want to keep the Loom consistent with the count_exons = true option?

File: pipelines/skylab/slideseq/SlideSeq.wdl [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+## This WDL pipeline implements data processing for RNA with UMIs
+##
+## Runtime parameters are optimized for Broad's Google Cloud Platform implementation.
+## For program versions, see docker containers.
+##
+## LICENSING :
+## This script is released under the WDL source code license (BSD-3) (see LICENSE in
+## https://github.com/broadinstitute/wdl). Note however that the programs it calls may
+## be subject to different licenses. Users are responsible for checking that they are
+## authorized to run all programs before running this script. Please see the docker
```
Comment (ekiernan) [inline pipelines/skylab/slideseq/SlideSeq.wdl:21]:
```suggestion
## authorized to run all programs before running this script. 
```

File: pipelines/skylab/slideseq/SlideSeq.wdl [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+## page at https://hub.docker.com/r/broadinstitute/genomes-in-the-cloud/ for detailed
```
Comment (ekiernan) [inline pipelines/skylab/slideseq/SlideSeq.wdl:22]:
```suggestion
```

File: pipelines/skylab/slideseq/SlideSeq.wdl [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+## licensing information pertaining to the included programs.
```
Comment (ekiernan) [inline pipelines/skylab/slideseq/SlideSeq.wdl:23]:
```suggestion
```

**Review by ekiernan — APPROVED** [review]

**Comment by nikellepetrillo** [comment]
retest this please

**Comment by nikellepetrillo** [comment]
plumbing test with count_exons = true : https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/42716/console

**Comment by nikellepetrillo** [comment]
retest this please

**Comment by nikellepetrillo** [comment]
scATAC is failing but the BAM files are the same, so we believe the failing tests are due to the non-determinism in calculating the fragments.csv

**Comment by nikellepetrillo** [comment]
retest this please

**Review by khajoue2 — APPROVED** [review]

File: pipelines/skylab/slideseq/test_inputs/test_data_overview.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# Slideseq Test Data
```
Comment (ekiernan) [inline pipelines/skylab/slideseq/test_inputs/test_data_overview.md:1]:
```suggestion
# SlideSeq Test Data
```

File: pipelines/skylab/slideseq/test_inputs/test_data_overview.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+The following lists the data used for workflow validation, including workflow engineering (plumbing) tests and scientific validation tests.
+
+All unmodified FASTQ files are stored in a public Google bucket (gs://nemo-public/biccn-unbundled/grant/rf1_macosko/macosko/spatial_transcriptome/cellgroup/Slide-seq/mouse/raw/) and can be identified using their sample ID (ex. Puck_210817_11).
+
+## Plumbing
+
+* Puck_210817_11.human
```
Comment (ekiernan) [inline pipelines/skylab/slideseq/test_inputs/test_data_overview.md:9]:
```suggestion
* Puck_210817_11.mm10
```

**Review by ekiernan — APPROVED** [review]

**Pull request merged**